rental_variant/models/sale_order.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# @author Kévin Roche <kevin.roche@akretion.com>
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        rentals = [line for line in self.order_line if line.is_rental]
+        for rental in rentals:
+            rented = (
+                rental.product_id.rented_product_id.id or rental.set_rented_product_id()
+            )
+            if not isinstance(rented, int):
+                rented = rented.id
+            rented_product = self.env["product.product"].browse(rented)
+            rented_product.rental_product_id = rental.product_id
+            rental.product_id.rented_product_id = rented_product
+            rental.rental_product_id_change()
+        return super().action_confirm()
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    rental_id = fields.Many2one(
+        comodel_name="sale.rental",
+        string="Rental",
+        compute="_compute_rental_id",
+        store=True,
+    )
+    is_rental = fields.Boolean(
+        string="Is Rental Variant Exist",
+        compute="_compute_is_rental",
+    )
+
+    @api.depends("product_id.product_tmpl_id")
+    def _compute_is_rental(self):
+        for rec in self:
+            rec.is_rental = bool(rec.product_id.product_tmpl_id.rented_product_tmpl_id)
+
+    @api.depends("order_id.state")
+    def _compute_rental_id(self):
+        for rec in self:
+            rec.rental_id = self.env["sale.rental"].search(
+                [("start_order_line_id", "=", rec.id)], limit=1
+            )
+
+    def set_rented_product_id(self):
+        if not self.product_id.rented_product_id:
+            rented_product_tmpl = self.product_id.product_tmpl_id.rented_product_tmpl_id
+            rented = rented_product_tmpl.product_variant_ids.filtered(
+                lambda x, s=self: (
+                    s.product_id.product_template_variant_value_ids.product_attribute_value_id
+                    == x.product_template_variant_value_ids.product_attribute_value_id
+                )
+            ).id
+            if not rented:
+                rented_ptav = self.env["product.template.attribute.value"].search(
+                    [
+                        ("product_tmpl_id", "=", rented_product_tmpl.id),
+                        (
+                            "product_attribute_value_id",
+                            "in",
+                            self.product_template_attribute_value_idsproduct_attribute_value_id.ids,
+                        ),
+                    ]
+                )
+
```

Comment:
To fix syntax error and avoid pre-commit fail
```suggestion
            if not rented:
                ptavs = self.product_template_attribute_value_ids
                rented_ptav = self.env["product.template.attribute.value"].search(
                    [
                        ("product_tmpl_id", "=", rented_product_tmpl.id),
                        (
                            "product_attribute_value_id",
                            "in",
                            ptavs.product_attribute_value_id.ids,
                        ),
                    ]
                )

```

Reply:
Thank you, it's done